### DISCUSSION TITLE
Getting info from draw on useControl

### DISCUSSION THREAD

**Discussion by zayne-anthony** [uber/react-map-gl#1949] [Q&A]
Hello, I am trying to add a feature to my map where you can isolate markers/points inside a drawn polygon. As of right now I am having trouble using the draw useControl. It is in the map just fine, but getting the values from onCreate/onUpdate is just not working. I have tried console logging any value I an think of when it should during the onCreate/onUpdate but I am not getting anything. Not even an error. It just seems it's not firing. I have followed the example on `react-map-gl` and even doing that it doesn't store anything inside the feature state.

Sandbox: https://codesandbox.io/s/condescending-margulis-yixddb?file=/src/Mapbox.js

Here is my code:
```jsx
// Map Component
const onCreate = useCallback((e) => {
     console.log(e.features);
}, []);

// Draw Component
import { useControl } from "react-map-gl";
import MapboxDraw from "@mapbox/mapbox-gl-draw";

const DrawControl = (props) => {
	useControl(
		({ map }) => {
			map.on("draw.create", props.onCreate);
			map.on("draw.update", props.onUpdate);
			map.on("draw.delete", props.onDelete);
			return new MapboxDraw(props);
		},
		({ map }) => {
			map.off("draw.create", props.onCreate);
			map.off("draw.update", props.onUpdate);
			map.off("draw.delete", props.onDelete);
		},
		{
			position: props.position,
		}
	);

	return null;
};

export default DrawControl;
```

**Comment by Pessimistress** [comment]
Create a CodeSandbox if you need help.

**Reply by zayne-anthony** [reply]
https://codesandbox.io/s/condescending-margulis-yixddb?file=/src/Mapbox.js

**Reply by Pessimistress** [reply]
If you remove `<StrictMode>` it works as expected. Can you create a bug?

**Reply by zayne-anthony** [reply]
Gotcha, thank you!

**Reply by Pessimistress** [reply]
This is fixed in v7.0.19, see the updated example in https://github.com/visgl/react-map-gl/blob/7.0-release/examples/draw-polygon/src/draw-control.ts

**Reply by engelzero** [reply]
@Pessimistress with that example with the useControl hook how can I get access to the DrawControl ref

`drawRef.current.add` to add a feature or change the select modes - something along those lines? Bit stuck as I'm migrating to latest dependency versions.

**Reply by norgepaul** [reply]
@engelzero I'm struggling with the same thing. I need the reference to add a feature for editing. Did you figure it out?

**Reply by engelzero** [reply]
@norgepaul https://github.com/visgl/react-map-gl/discussions/1770#discussioncomment-3546189 - sure did!

**Reply by joel-daros** [reply]
@Pessimistress Could you post a simple example how to call these [API methods ](https://github.com/mapbox/mapbox-gl-draw/blob/main/docs/API.md#api-methods) from the `<DrawControl>`

**Reply by Thormod** [reply]
Same problem here!
